### ISSUE TITLE
ActiveStorage::MirrorJob always uses the default service, even if overridden via `has_*_attached ..., service: :another_service`

### ISSUE THREAD

**Issue by jnimety** [rails/rails#46806]
I ran into this when trying to migrate all of my attached from local storage to s3. After setting up mirroring and running `ActiveStorage::Blob.find_each &:mirror_later` I was surprised to find all of my attachments in the default storage service, even if a different service had been specified via `has_*_attached ..., service: ...`

I think the culprit is here:

https://github.com/rails/rails/blob/a7902034089e8b6bff747c08d93eeac4b1377032/activestorage/app/jobs/active_storage/mirror_job.rb#L13

`ActiveStorage::Blob.service` the default service. The other jobs pass `blob` to perform... Is there any reason not to pass `blob` here as well, and use `blob.service` instead of `ActiveStorage::Blob.service`?

If that's acceptable I'm happy to submit a PR. Thanks!

**Comment by ghiculescu** [comment]
Yep that looks like a bug. Please submit a PR! You could add a test to verify the fix [here](https://github.com/rails/rails/blob/40a1f3ae24b297cddcfca06332fe113faf2ab52f/activestorage/test/models/attachment_test.rb#L41).

**Comment by jnimety** [comment]
I'm unfamiliar with the rails PR process... is there anything else I need to do to move this forward?

**Comment by skipkayhil** [comment]
Your best bet is probably asking for a review in the [Contributor Discord](https://rubyonrails.org/2022/6/13/rails-discord-server-is-now-open-to-the-public)
